### PR TITLE
[RocksJava] Promote CompactionFilter* accessors to ColumnFamilyOptionsInterface

### PR DESCRIPTION
**Pull request by benclay**
When adding CompactionFilter and CompactionFilterFactory settings to the Java layer, ColumnFamilyOptions was modified directly instead of ColumnFamilyOptionsInterface. This meant that the old-stye Options monolith was left behind.  

This patch fixes that, by:
- promoting the CompactionFilter + CompactionFilterFactory setters from ColumnFamilyOptions -> ColumnFamilyOptionsInterface
- adding getters in ColumnFamilyOptionsInterface
- implementing setters in Options
- implementing getters in both ColumnFamilyOptions and Options
- adding testcases
- reusing a test CompactionFilterFactory by moving it to a common location

### CODE REVIEW

**Comment by benclay** [comment]
@sagar0 found this trolling through some old stuff.  Any chance of a review?

File: java/src/test/java/org/rocksdb/test/TestCompactionFilterFactory.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package org.rocksdb.test;
+
+import org.rocksdb.AbstractCompactionFilter;
+import org.rocksdb.AbstractCompactionFilterFactory;
+import org.rocksdb.RemoveEmptyValueCompactionFilter;
+
+/**
+ * Simple CompactionFilterFactory class used in tests. Generates RemoveEmptyValueCompactionFilters.
+ */
+public class TestCompactionFilterFactory extends AbstractCompactionFilterFactory<RemoveEmptyValueCompactionFilter> {
```
[the remark this answers is not in the document]
Comment (benclay) [reply java/src/test/java/org/rocksdb/test/TestCompactionFilterFactory.java]: Sure, done.

**Comment by sagar0** [comment]
Fixed merge conflicts.

**Comment by sagar0** [comment]
Fixed more merge conflicts due to the just landed PR #4764.

**Comment by benclay** [comment]
@sagar0 thanks for babysitting this.

**Pull request closed without merging**